storage/series/buffer.go

```diff
@@ -739,13 +745,20 @@ func (b *dbBufferBucket) merge() (mergeResult, error) {
 		}
 	}
 
-	readers := make([]io.Reader, 0, len(b.encoders)+len(b.bootstrapped))
-	streams := make([]xio.SegmentReader, 0, len(b.encoders))
+	start := b.start
+	end := start.Add(b.opts.RetentionOptions().BlockSize())
+	readers := make([]xio.SegmentReader, 0, len(b.encoders)+len(b.bootstrapped))
+	streams := make([]xio.BlockReader, 0, len(b.encoders))
 	for i := range b.encoders {
-		if stream := b.encoders[i].encoder.Stream(); stream != nil {
+		if s := b.encoders[i].encoder.Stream(); s != nil {
 			merges++
-			readers = append(readers, stream)
-			streams = append(streams, stream)
+			br := xio.BlockReader{
+				SegmentReader: s,
+				Start:         start,
+				End:           end,
+			}
+			readers = append(readers, br)
```

Comment:
Here too, can you `append(readers, s)`?  Will be the same result as you are merging just below.  

In fact probably don't need the `BlockReader` at all here.